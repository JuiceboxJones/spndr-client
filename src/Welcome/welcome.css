.wishlist_window{
  width: 33%;
  margin-top: 5%;
  border-radius: 8px;
  background: #f5f5f5;
  padding: 10px;
  float: right;
  height: 50%;
  position: relative;
  margin-right: 10%;
}

.wishlist_window li{
  border-radius: 2px;
  background: #dbdada;
  font-size: 17px;
  font-weight: bold;
}

.wishlist_window li span{
  margin-left: 20px;
}

#logout_button{
  margin: 5px;
  position: absolute;
  top: 20px;
  right: 25px;
  border: none;
  width: 150px;
  height: 25px;
  border-radius: 5px;
  background: #024e2e9c;
  color: white;
  cursor: pointer;
  font-weight: bold;
}

#red_total{
  color: rgb(209, 0, 0);
}

#logout_button:hover{
  background: rgb(122, 155, 143);
  color: rgb(209, 0, 0);
}


#new_budget_button{
  margin: 15px;
  left: 37%;
  border: none;
  width: 150px;
  height: 25px;
  border-radius: 5px;
  background:#A7C4C0;
  cursor: pointer;
  font-weight: bold;
}

#new_budget_button:hover{
  background: #024e2e9c;
  color:white;
}

#edit{
  position: relative;
  margin: 0 auto;
  border: none;
  width: 150px;
  height: 25px;
  border-radius: 5px;
  background:#A7C4C0;
  cursor: pointer;
  font-weight: bold;
}
#edit:hover{ 
  background: #024e2e9c;
  color:white;
  }

.current_limit{
  width: 33%;
  margin-top: 5%;
  border-radius: 8px;
  background: #f5f5f5;
  padding: 10px;
  float: left;
  height: 150px;
  margin-left: 33%;
}

.expenses_window{
  width: 33%;
  margin-top: 5%;
  border-radius: 8px;
  background: #f5f5f5;
  padding: 10px;
  float: left;
  height: 50%;
  margin-left: 10%;
}

.expenses_window li{
  border-radius: 2px;
  background: #dbdada;
  font-size: 17px;
  height: 25px;
  font-weight: bold;
  color: rgb(209, 0, 0);
}

.expenses_window li span{
  margin-left: 20px;

}


.expenses_window p {
  text-align: center;
  margin-left: 15%;
}

#delete_expense{
  float: right;
  margin-right: 20px;
  margin-top: 2px;
  border: none;
  background: none;
  cursor: pointer;
}
#delete_expense:hover{
  font-size: 105%;
  text-shadow: 7px -3px 3px rgba(0, 0, 0, 0.644);
}

@media screen and (max-width: 480px) {
  .current_limit {
    width: 50%;
    margin-left: 25%;
    height: 130px;
    margin-bottom: 18px;
  }
  #new_budget_button{
    margin: 3px;
  }
  .expenses_window{
    margin-left: 2%;
    width: 42%;
    padding: 8px;
  }
  .wishlist_window{
    margin-right: 2%;
    width: 42%;
  } 
  .expenses_window p{
    font-size: 14px;
    margin-left: 2px;
  }
  .expenses_window li span{
    margin-left: 5px;
  }
  #delete{
    margin-right: 0px;
    margin-top: -1px;
  }
  #delete_expense{
    margin-right: 8px;
  }
  h3{
    font-size: 15px;
  }
  #logout_button{
    top: 90px;
    right: 86px;
  }
  .wishlist_window li{
    font-size: 15px;
  }
  .wishlist_window li span{
    margin-left: 4px;
  }
}

@media screen and (max-width: 320px) {

  .current_limit{
    margin-left: 21%;
  }
  .current_limit h3{
    margin: 5px;
  }
}

@media screen and (max-width: 330px){
  .current_limit{
    margin-top: 19px;
  }
  #edit{
    width: 135px;
  }
  #delete_expense{
    margin-right: -8px;
    margin-top: 0px;
  }
}

