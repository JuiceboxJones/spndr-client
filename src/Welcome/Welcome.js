import React, { Component } from 'react';
import ApiService from '../services/api-fetch-services';
import TokenService from '../services/token-service';
import HistoryHelper from '../History/History';
import Header from '../Header/Header';
import './welcome.css'

//current date without time  new Date().toISOString().slice(0,10)

class Welcome extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {}
    }
  };

  state = {
    expensesTotal: null,
    expenses: [],
    income: null,
    wishlist: [],
    spendLimit: null,
    editWishlist: false,
    editExpenses: false
  };

  componentDidMount() {
    Promise.all([
      ApiService.getExpenses(),
      ApiService.getIncome(),
      ApiService.getWishlist()
    ]).then(data =>
      this.setState(
        {
          expenses: data[0],
          income: data[1],
          wishlist: data[2]
        },
        () => this.handleExpensesTotal(this.state.expenses)
      )
    );
  }


  handleExpensesTotal(exp) {
    if (exp.length !== 0) {
      const newArr = [];
      for (let i = 0; i < exp.length; i++) {
        newArr.push(Number(exp[i].amount));
      }
      const presum = newArr.reduce((a, b) => a + b);
      const sum = presum.toFixed(2)
      this.setState({ expensesTotal: sum }, () =>
        this.handleIncome(this.state.income)
      );
    }
  }

  handleDelete(e) {
    ApiService.deleteExpense(e);
    this.handleDeleteitem(e);
  }

  handleDeleteitem = e => {
    const expenses = this.state.expenses.filter(item => item.id !== e);
    this.setState(
      { expenses },
      () => this.handleExpensesTotal(this.state.expenses),
      this.handleIncome(this.state.income)
    );
  };

  handleIncome(inc) {
    if (inc.length !== 0) {
      const subtotal = inc[0].income - inc[0].add_savings;
      const total = subtotal - this.state.expensesTotal;
      const limit = total.toFixed(2);
      this.setState({ spendLimit: limit });
    }
  }

  handleLogout = e => {
    TokenService.clearAuthToken();
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/';
    history.push(destination);
  };

  handleButtonNav(page) {
    HistoryHelper.historyGoTo(page, this.props);
  }

  render() {
    const handleWishlist = this.state.wishlist.map(wishes => {
      return (
        <li key={wishes.id}>
          <a href={wishes.url}>{wishes.name}</a><span/> ${wishes.price}
          <p />
        </li>
      );
    });

    const handleExpenseList = this.state.expenses.map(exp => {
      return (
        <li key={exp.id}>
          <p>
             {exp.name}<span />${exp.amount}
            <button id='delete_expense' type="button" onClick={() => this.handleDelete(exp.id)}>
            <span id='delete' role="img" aria-label="delete">‚ùå</span>
            </button>{' '}
          </p>
        </li>
      );
    });

    return (
      <div>
        <header>
          <Header />
        </header>
        <div className="current_limit">
          <h3>
            Currently, your monthly spending budget is: 
            <p/>${this.state.spendLimit}
          </h3>
          <button type="button" id='new_budget_button' onClick={() => this.handleButtonNav('/income')}>
            New Budget
          </button>
        </div>
        <div className="wishlist_window">
          <h3>My Wishlist</h3>
          <ul>{handleWishlist}</ul>
          <button type="button" id='edit' onClick={() => this.handleButtonNav('/wishlist')}>Edit Wishlist</button>
        </div>
        <div className="expenses_window">
          <h3>Current Expenses</h3>
          <ul>{handleExpenseList}</ul>
          <h4>Total: <span id='red_total'>${this.state.expensesTotal}</span></h4>
        </div>
        <div className="button_menu">
          <button type="button" id='logout_button' onClick={e => this.handleLogout(e)}>
            Logout
          </button>
        </div>
      </div>
    );
  }
}

export default Welcome;
