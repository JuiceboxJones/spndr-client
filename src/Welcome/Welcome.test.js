import React from 'react';
import ReactDOM from 'react-dom';
import Welcome from './Welcome'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Welcome />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Welcome />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});