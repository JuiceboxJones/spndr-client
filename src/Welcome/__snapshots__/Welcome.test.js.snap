// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header>
    <div
      className="header"
    >
      <header>
        <h1>
          Welcome to SPNDR
        </h1>
      </header>
      <p>
        Finance Management. Simplified.
      </p>
    </div>
  </header>
  <div
    className="current_limit"
  >
    <h3>
      Currently, your monthly spending budget is:
      <p />
      $
    </h3>
    <button
      id="new_budget_button"
      onClick={[Function]}
      type="button"
    >
      New Budget
    </button>
  </div>
  <div
    className="wishlist_window"
  >
    <h3>
      My Wishlist
    </h3>
    <ul />
    <button
      id="edit"
      onClick={[Function]}
      type="button"
    >
      Edit Wishlist
    </button>
  </div>
  <div
    className="expenses_window"
  >
    <h3>
      Current Expenses
    </h3>
    <ul />
    <h4>
      Total: 
      <span
        id="red_total"
      >
        $
      </span>
    </h4>
  </div>
  <div
    className="button_menu"
  >
    <button
      id="logout_button"
      onClick={[Function]}
      type="button"
    >
      Logout
    </button>
  </div>
</div>
`;
